impala/tests/test_data_types.py

```diff
@@ -69,6 +69,32 @@ def test_date_basic(cur, date_table):
     results = cur.fetchall()
     assert results == [(datetime.date(1, 1, 1),), (datetime.date(1999, 9, 9),)]
 
+
+@yield_fixture(scope='module')
+def timestamp_table(cur):
+    table_name = 'tmp_timestamp_table'
+    ddl = """CREATE TABLE {0} (ts timestamp)""".format(table_name)
+    cur.execute(ddl)
+    try:
+        yield table_name
+    finally:
+        cur.execute("DROP TABLE {0}".format(table_name))
+
+
+@pytest.mark.connect
+def test_timestamp_basic(cur, timestamp_table):
```

Comment:
Can you also test timestamps with sub-second component? E.g. something like 
select cast("2021-05-20 00:00:00.123 " as timestamp), 
          cast("2021-05-20 00:00:00.123456 " as timestamp)
          cast("2021-05-20 00:00:00.123456789 " as timestamp);

datetime.datetime should be able to store them with microseconds precision, so the last one should not return the exact value.